packages/entity-server/src/routes/hierarchy/middleware.ts

```diff
@@ -46,39 +52,44 @@ export const attachFormatterToResponse = async (
   }
 };
 
-const userHasAccessToEntity = async (
-  entity: EntityModel,
-  projectCode: string,
+const userHasAccessToProject = async (
+  models: EntityServerModelRegistry,
+  project: ProjectType,
   accessPolicy: AccessPolicy,
 ) => {
-  // Assume user always has access to all world items.
-  if (entity.code === 'World') {
-    return true;
-  }
+  const projectEntity = await models.entity.findOne({ code: project.code });
+  const projectChildren = await projectEntity.getChildrenViaHierarchy(project.entity_hierarchy_id);
+
+  return accessPolicy.allowsSome(projectChildren.map(c => c.country_code));
+};
+
+const userHasAccessToEntity = async (entity: EntityType, accessPolicy: AccessPolicy) => {
   // Timor-Leste is temporarily turned off
+  // Note: Remove this check as part of: https://github.com/beyondessential/tupaia-backlog/issues/2456
   if (entity.country_code === 'TL') {
     return false;
   }
 
-  // project access rights are determined by their children
   if (entity.isProject()) {
-    const projectChildren = await entity.getChildren(projectCode);
-
-    return accessPolicy.allowsSome(projectChildren.map(c => c.country_code));
+    return true; // Already checked this in userHasAccessToProject
   }
 
   return accessPolicy.allows(entity.country_code);
 };
 
-const mapEntitiesToFields = (fields: (keyof EntityFields)[]) => (model: EntityModel) => {
+const mapEntityToFields = (fields: (keyof EntityFields)[]) => (entity: EntityType) => {
   const mappedEntity: { -readonly [field in keyof EntityFields]?: EntityFields[field] } = {};
   fields.forEach(field => {
-    mappedEntity[field] = model[field];
+    mappedEntity[field] = entity[field];
   });
   return mappedEntity;
 };
 
 const validFields: (keyof EntityFields)[] = ['code', 'country_code'];
+const validateField = (field: string): field is keyof EntityFields => {
```

Comment:
Thanks for moving this one, but it was actually a general comment for throughout this file 🙏

Reply:
Oh I see. I think my desire to have exported members at the top was conflicting with this. Will fix the rest of the file 👍